controllers/configurationpolicy_utils.go

```diff
@@ -577,98 +577,161 @@ func createStatus(
-			case reasonUpdateSuccess:
-				generatedReason = reasonUpdateSuccess
-				generatedMsg = fmt.Sprintf("%s%s was updated successfully", resourceName, namesStr)
-			case reasonDeleteSuccess:
-				generatedReason = reasonDeleteSuccess
-				generatedMsg = fmt.Sprintf("%s%s was deleted successfully", resourceName, namesStr)
-			case reasonWantFoundDNE:
-				generatedReason = "K8s does not have a `must have` object"
-				compliancyDetailsMsg += fmt.Sprintf("%s%s not found", resourceName, namesStr)
-			case reasonWantFoundNoMatch:
-				generatedReason = "K8s does not have a `must have` object"
-				compliancyDetailsMsg += fmt.Sprintf("%s%s found but not as specified", resourceName, namesStr)
-			case reasonWantNotFoundExists:
-				generatedReason = "K8s has a `must not have` object"
-				compliancyDetailsMsg += fmt.Sprintf("%s%s found", resourceName, namesStr)
-			case reasonWantNotFoundDNE:
-				generatedReason = "K8s `must not have` object already missing"
-				compliancyDetailsMsg += fmt.Sprintf("%s%s missing as expected", resourceName, namesStr)
-			default:
-				// If it's not one of the above reasons, then skip consolidation. This is likely an error being
-				// reported.
-				if i == 0 {
-					if compliancyDetailsReason != "" {
-						compliancyDetailsReason += "; "
-					}
-
-					compliancyDetailsReason += reason
-				}
+		msgMap, compliancyDetailsReason, compliancyDetailsMsg = generateMessageWithReason(sortedObjectNamesStrs,
+			objectNameStrsToNsNames, nsNameToEvent,
+			compliancyDetailsReason, compliancyDetailsMsg, reason)
 
-				for j, nsName := range objectNameStrsToNsNames[namesStr] {
-					if j != 0 {
-						compliancyDetailsMsg += "; "
-					}
+		compliancyDetailsMsg = getCombinedCompliancyDetailsMsg(msgMap, resourceName, compliancyDetailsMsg)
+	}
 
-					compliancyDetailsMsg += nsNameToEvent[nsName].event.message
-				}
+	return
+}
 
-				// Assume the included messages include the namespace.
-				continue
-			}
+func setCompliancyDetailsMsgEnd(compliancyDetailsMsg string) string {
+	if compliancyDetailsMsg != "" && !strings.HasSuffix(compliancyDetailsMsg, "; ") {
+		compliancyDetailsMsg += "; "
+	}
+
+	return compliancyDetailsMsg
+}
+
+func getCombinedCompliancyDetailsMsg(msgMap map[string][]string,
+	resourceName string, compliancyDetailsMsg string,
+) string {
+	// Sort msgTemplate keys for consistent processing
+	msgTemplates := make([]string, 0, len(msgMap))
+	for k := range msgMap {
+		msgTemplates = append(msgTemplates, k)
+	}
+
+	slices.Sort(msgTemplates)
+
+	for _, msgTemplate := range msgTemplates {
+		names := msgMap[msgTemplate]
+		slices.Sort(names) // Sort names for consistent output
+
+		compliancyDetailsMsg = setCompliancyDetailsMsgEnd(compliancyDetailsMsg)
+
+		joinedNames := strings.Join(names, ", ")
+		if joinedNames != "" {
+			joinedNames = fmt.Sprintf(" [%s]", joinedNames)
+		}
+
+		compliancyDetailsMsg += fmt.Sprintf(msgTemplate, resourceName, joinedNames)
+	}
```

Comment:
Assuming the index approach works, I think that would be more clear here as well:
```suggestion
	for j, msgTemplate := range msgTemplates {
		names := msgMap[msgTemplate]
		slices.Sort(names) // Sort names for consistent output

		if j != 0 {
			compliancyDetailsMsg += "; "
		}
		
		joinedNames := strings.Join(names, ", ")
		if joinedNames != "" {
			joinedNames = fmt.Sprintf(" [%s]", joinedNames)
		}

		compliancyDetailsMsg += fmt.Sprintf(msgTemplate, resourceName, joinedNames)
	}
```